### PR TITLE
feat: Upgrade create-llama and add new tools

### PR DESCRIPTION
**Pull request by leehuwuj**

### CODE REVIEW

File: src/ragapp/Makefile [hunk 1]
```diff
@@ -60,3 +61,7 @@ dev:
 	poetry run python main.py & \
 	pnpm --prefix ./admin-ui run dev & \
 	wait
+
+mypy:
```
Comment (marcusschiesser) [inline src/ragapp/Makefile:65]: do we trigger mypy in CI?

Comment (leehuwuj) [reply src/ragapp/Makefile:65]: Ah, sorry. We don't trigger mypy in the CI as it's not working for RAGapp codebase now. I tried to add mypy to RAGapp but it had conflict errors that i couldn't fix it quickly. I'll work on this later.

File: src/ragapp/templates/financial_analyst.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+name: FinancialAnalyst
+role: Financial Analyst
+backstory: Expert in analyzing financial information and able to execute code to provided data.
```
Comment (marcusschiesser) [inline src/ragapp/templates/financial_analyst.yaml:3]:
```suggestion
backstory: Expert in analyzing financial information. Able to execute code to derive new insights for provided data.
```

File: src/ragapp/patch/frontend/app/components/ui/chat/chat-message/markdown.tsx [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+            const fileExtension = href.split(".").pop()?.toLowerCase();
+
+            if (
+              fileExtension &&
+              DOCUMENT_FILE_TYPES.includes(fileExtension as DocumentFileType)
+            ) {
+              return (
+                <DocumentInfo
+                  document={{
+                    url: new URL(decodeURIComponent(href), window.location.origin).href,
```
Comment (marcusschiesser) [inline src/ragapp/patch/frontend/app/components/ui/chat/chat-message/markdown.tsx:138]:
```suggestion
                    url: decodeURIComponent(href),
```

**Review by marcusschiesser — APPROVED** [review]

**Pull request merged**